src/components/TodoApp.tsx

```diff
@@ -0,0 +1,221 @@
+    setLocalTodos([...localTodos, todo]);
+  };
+
+  useEffect(() => {
+    if (newTodoField.current) {
+      newTodoField.current.focus();
+    }
+
+    if (user) {
+      getTodos(user.id)
+        .then((loadedTodos) => {
+          setTodos(loadedTodos);
+        })
+        .catch(() => {
+          setErrorMessage('Can\'t load todos');
+          setTimeout(handleErrorClose, 3000);
+        });
+    }
+  }, []);
+
+  const addNewTodo = async (title: string) => {
+    if (!title.trim()) {
+      setErrorMessage(Error.EMPTY);
+      setTimeout(handleErrorClose, 3000);
+
+      return;
+    }
+
+    if (user) {
+      try {
+        setTempTodo({
+          id: 0,
+          title,
+          userId: user.id,
+          completed: false,
+        });
+
+        const newTodo = await addTodo(user.id, title);
+
+        setTodos(allTodos => [...allTodos, newTodo]);
+      } catch {
+        setErrorMessage(Error.ADD);
+      } finally {
+        setTempTodo(null);
+      }
+    }
+  };
+
+  const removeTodo = async (todoId: number) => {
+    setIdsToChange(current => [...current, todoId]);
+
+    if (user) {
+      try {
+        await deleteTodo(todoId);
+
+        setTodos(allTodos => allTodos.filter(
+          todo => todo.id !== todoId,
+        ));
+      } catch {
+        setErrorMessage(Error.DELETE);
+      } finally {
+        setIdsToChange([]);
+      }
+    }
+  };
+
+  const updateStatusTodo = async (todo: Todo) => {
+    setIdsToChange(current => [...current, todo.id]);
+
+    if (user) {
+      try {
+        const updatedTodo = { ...todo, completed: !todo.completed };
+
+        await updateTodo(todo.id, { completed: !todo.completed });
+        setTodos(allTodos => allTodos
+          .map(element => {
+            if (element.id === todo.id) {
+              return updatedTodo;
+            }
+
+            return element;
+          }));
+      } catch {
+        setErrorMessage(Error.UPDATE);
+      } finally {
+        setIdsToChange([]);
+      }
+    }
+  };
+
+  const editTitleTodo = async (todo: Todo, title: string) => {
+    setIdsToChange(current => [...current, todo.id]);
+
+    if (user) {
+      try {
+        const updatedTodo = { ...todo, title };
+
+        await updateTodo(todo.id, { title });
+        setTodos(allTodos => allTodos
+          .map(element => {
+            if (element.id === todo.id) {
+              return updatedTodo;
+            }
+
+            return element;
+          }));
+      } catch {
+        setErrorMessage(Error.UPDATE);
+      } finally {
+        setIdsToChange([]);
+      }
+    }
+  };
+
+  const allProps = {
+    onTodoDelete: removeTodo,
+    tempTodo,
+    updateStatusTodo,
+    idsToChange,
+    editTitleTodo,
+    localTodo,
+  };
+
+  const activeTodos = todos.filter(todo => !todo.completed);
+  const completedTodos = todos.filter(todo => todo.completed);
+
+  const isAllTodosCompleted: boolean = todos.length === completedTodos.length;
```

Comment:
Typescript should understand that it's boolean type and developers will understand this by `is` prefix:
```suggestion
  const isAllTodosCompleted = todos.length === completedTodos.length;
```